src/python.cc

```diff
@@ -799,6 +799,7 @@ ModelInstanceState::SetupChildProcess()
   }
 
   parent_pid_ = getpid();
+  pthread_mutex_lock(parent_mutex_);
```

Comment:
InstanceFinalize is not called unless InstanceInitialize returns with a non-error code. Will the parent mutex be left in a locked state if initialization fails? Does that matter?

Reply:
The parent will not be in the locked state because the wait for the condition variable is timed. It will wait for some default value until the stub notifies the parent process. The problem here was that the parent didn't lock the mutex but child notified the parent and the notification was lost.